pkg/controller/volume/attachdetach/util/util.go

```diff
@@ -159,6 +159,23 @@ func getPVSpecFromCache(name string, pvcReadOnly bool, expectedClaimUID types.UI
 	return volume.NewSpecFromPersistentVolume(clonedPV, pvcReadOnly), nil
 }
 
+// DetermineVolumeAction returns true if volume and pod needs to be added to dswp
+// and it returns false if volume and pod needs to be removed from dswp
+func DetermineVolumeAction(pod *v1.Pod, desiredStateOfWorld cache.DesiredStateOfWorld, defaultValue bool) bool {
```

Comment:
What you mean - we already know the action? The volume action (attach or detach) is decided not only by state of the pod (terminated or not) but also if user has asked kubelet to keep terminated volumes around. 

I am happy to change function name though. The only thing I don't like about this function is - it returns a boolean variable whereas it sounds like it is going to return name of the action (`attach` or `detach`). So this function is bit like a gate/latch - that determines which way the code will flow.